hierarchicalforecast/core.py

```diff
@@ -244,17 +244,16 @@ def reconcile(self,
                 S_for_sparse = sparse.csr_matrix(S_df.sparse.to_coo())
             except AttributeError:
                 warnings.warn('Using dense S matrix for sparse reconciliation method.')
-                S_for_sparse = S_df.values.astype(np.float32)
+                S_for_sparse = S_df.values.astype(np.float64)
```

Comment:
I'd add `copy=False` in these in case the array is already double

Reply:
Good point